src/formatDate.js

```diff
@@ -8,7 +8,61 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat[fromFormat.length - 1];
+  const toSeparator = toFormat[toFormat.length - 1];
+  const dateParts = date.split(`${fromSeparator}`);
+  let year = '';
+  let month = '';
+  let day = '';
+  const result = [];
+
+  for (let i = 0; i <= dateParts.length; i++) {
+    if (fromFormat[i] === 'YYYY') {
+      year = dateParts[i];
+    }
+
+    if (fromFormat[i] === 'YY') {
+      year = dateParts[i];
```

Comment:
When converting from 'YY' to a full year, you need to determine whether to prepend '19' or '20' to the year. This logic is missing here and should be handled similarly to lines 42-48.